aderyn_core/src/report/markdown_printer.rs

```diff
@@ -0,0 +1,222 @@
+        )?;
+        Ok(())
+    }
+    fn print_contract_summary<W: Write>(
+        &self,
+        mut writer: W,
+        report: &Report,
+        loader: &ContextLoader,
+    ) -> Result<()> {
+        writeln!(writer, "# Summary\n")?;
+
+        // Files Summary
+        {
+            writeln!(writer, "## Files Summary\n")?;
+            let files_summary = loader.files_summary();
+
+            // Start the markdown table
+            writeln!(writer, "| Key | Value |")?;
+            writeln!(writer, "| --- | --- |")?;
+            writeln!(
+                writer,
+                "| .sol Files | {} |",
+                files_summary.total_source_units
+            )?;
+            writeln!(writer, "| Total nSLOC | {} |", files_summary.total_sloc)?;
+
+            writeln!(writer, "\n")?; // Add an extra newline for spacing
+        }
+
+        // Files Details
+        {
+            writeln!(writer, "## Files Details\n")?;
+
+            // Start the markdown table with the header
+            writeln!(writer, "| Filepath | nSLOC |")?;
+            writeln!(writer, "| --- | --- |")?;
+
+            let files_details = loader.files_details();
+
+            files_details.files_details.iter().for_each(|detail| {
+                writeln!(writer, "| {} | {} |", detail.file_path, detail.n_sloc).unwrap();
+            });
+
+            let sloc_stats = &loader.sloc_stats;
+            writeln!(writer, "| **Total** | **{}** |", sloc_stats.code)?;
+            writeln!(writer, "\n")?; // Add an extra newline for spacing
+        }
+
+        // Analysis Sumarry
+        {
+            let issue_count = report.issue_count();
+            writeln!(writer, "## Issue Summary\n")?;
+
+            // Start the markdown table
+            writeln!(writer, "| Category | No. of Issues |")?;
+            writeln!(writer, "| --- | --- |")?;
+            writeln!(writer, "| Critical | {} |", issue_count.critical)?;
+            writeln!(writer, "| High | {} |", issue_count.high)?;
+            writeln!(writer, "| Medium | {} |", issue_count.medium)?;
+            writeln!(writer, "| Low | {} |", issue_count.low)?;
+            writeln!(writer, "| NC | {} |", issue_count.nc)?;
+            writeln!(writer, "\n")?; // Add an extra newline for spacing
+        }
+
+        Ok(())
+    }
+
+    fn print_table_of_contents<W: Write>(&self, mut writer: W, report: &Report) -> Result<()> {
+        fn print_table_of_content<T>(issues: &[Issue], mut w: T)
+        where
+            T: Write,
+        {
+            for (index, issue) in issues.iter().enumerate() {
+                let issue_title_slug = issue
+                    .title
+                    .to_lowercase()
+                    .replace(' ', "-")
+                    .replace(|c: char| !c.is_ascii_alphanumeric() && c != '-', "");
+                writeln!(
+                    w,
+                    "  - [C-{}: {}](#C-{}-{})",
```

Comment:
This currently prints `C` for all severities.

Reply:
@alexroan - I believe not having lowercase is why report tests are failing.